src/schema/schema_manager.py

```diff
@@ -697,6 +838,104 @@ def normalize_entity_result_for_response(entity_dict, properties_to_exclude = []
 
+"""
+Normalize the entity result to got into the OpenSearch index by filtering out properties that are not defined in
+the yaml schema, properties marked as `exposed: false` in the yaml schema, and properties lacking `indexed: true`
+marking in the yaml schema.
+
+Parameters
+----------
+entity_dict : dict
+    Either a neo4j node converted dict or metadata dict generated from get_index_metadata()
+properties_to_exclude : list
+    Any additional properties to exclude from the response
+
+Returns
+-------
+dict
+    An entity metadata dictionary with keys that are all normalized
+"""
+def normalize_document_result_for_response(entity_dict, properties_to_exclude=[]):
+    return _normalize_metadata( entity_dict=entity_dict
+                                , metadata_scope=MetadataScopeEnum.INDEX
+                                , properties_to_skip=properties_to_exclude)
+
+"""
+Normalize the entity result by filtering the properties to those appropriate for the
+scope of metadata requested e.g. complete data for a another service, indexing data for an OpenSearch document, etc.
+
+Properties that are not defined in the yaml schema and properties marked as `exposed: false` in the yaml schema are
+removed. Properties are also filter based upon the metadata_scope argument e.g. properties lacking `indexed: true`
+marking in the yaml schema are removed when `metadata_scope` has a value of `MetadataScopeEnum.INDEX`.
+
+Parameters
+----------
+entity_dict : dict
+    Either a neo4j node converted dict or metadata dict generated from get_index_metadata()
+metadata_scope:
+    A recognized scope from the SchemaConstants, controlling the triggers which are fired and elements
+    from Neo4j which are retained.  Default is MetadataScopeEnum.INDEX.
+properties_to_exclude : list
+    Any additional properties to exclude from the response
+
+Returns
+-------
+dict
+    An entity metadata dictionary with keys that are all normalized appropriately for the metadata_scope argument value.
+"""
+def _normalize_metadata(entity_dict, metadata_scope:MetadataScopeEnum, properties_to_skip=[]):
+    global _schema
+
+    # When the entity_dict is unavailable or the entity was incorrectly created, do not
+    # try to normalize.
+    if not entity_dict or 'entity_type' not in entity_dict:
+        return {}
+
+    normalized_metadata = {}
+
+    normalized_entity_type = entity_dict['entity_type']
+    properties = _schema['ENTITIES'][normalized_entity_type]['properties']
+
+    for key in entity_dict:
+        # Only return the properties defined in the schema yaml
+        # Exclude additional properties if specified
+        if  key in properties and \
+            key in properties_to_skip:
+            # Skip properties if directed by the calling function
+            continue
+        if  entity_dict[key] is None:
+            # Do not include properties in the metadata if they are empty
+            continue
+        if  'exposed' in properties[key] and \
```

Comment:
This will cause `KeyError` issues when a key returned by the neo4j query is not defined in the schema yaml, and we have such fields. There should be an outer "if check" to restrict the scope. 

For example, line 899 can use 
```
if key in entity_dict and key in properties:
```